integration-tests/src/test/java/oracle/kubernetes/operator/ItDomainInImage.java

```diff
@@ -123,6 +123,51 @@ public void testDomainInImageUsingWlst() throws Exception {
     LoggerHelper.getLocal().log(Level.INFO, "SUCCESS - " + testMethodName);
   }
 
+
+  /**
+   * Creates a WebLogic domain using domain-in-image option. Verifies the domain is started
+   * successfully and the SSL listeners in the WebLogic servers are active.
+   *
+   * @throws Exception If an error occurred
+   */
+  @Test
+  public void testDomainInImageUsingWlstWithSSLEnabled() throws Exception {
+    Assumptions.assumeTrue(FULLTEST);
+    String testMethodName = new Object() {
+    }.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    LoggerHelper.getLocal().log(Level.INFO, "Creating Domain & verifying the domain creation");
+    // create domain
+    Domain domain = null;
+    boolean testCompletedSuccessfully = false;
+    try {
+      Map<String, Object> domainMap = createDomainInImageMap(
+              getNewSuffixCount(), false, testClassName);
+      domainMap.put("namespace", domainNS1);
+      domainMap.remove("clusterType");
+
+      // domainMap key/value pairs are hard coded in the create domain method call above
+      domainMap.put("sslEnabled", Boolean.TRUE);
+      domainMap.put("javaOptions",
+              "-Dweblogic.StdoutDebugEnabled=false -Dweblogic.security.SSL.ignoreHostnameVerification=true");
+
+      domain = TestUtils.createDomain(domainMap);
+      domain.verifyDomainCreated();
+      domain.verifySSLListeners();
+      testCompletedSuccessfully = true;
```

Comment:
i would really like to see some assertions in here please

Reply:
@markxnelson I'm not sure what this comment implies? Are you talking JUnit assertions? JMockit style assertions? Or some other assertions? I may have missed something new here?

Reply:
I don't know if there's any additional assurance to have assertions after line 157.  Any of those domain creation, domain verification, SSL listener verification steps throws an exception if it encounters an error.  I can refactor the code to obtain the exec result and do some kind of assertion here but that seems a bit redundant in my opinion.

Reply:
@rjeberhard @ddsharpe what do you think?
i do not like the style of relying on an exception being thrown. if you think there may be one, i prefer to explicitly catch it and write an assertion about whether it was/was not thrown, its type, etc.
i think the real issue is that someone could independently change any of these methods, and maybe they no longer throw an exception, and now this test would be broken, but we would not know.

Reply:
@qvdang @TheFrogPad @markxnelson I would prefer NOT to rely on Exceptions in this way.  Yes, this is what we did in JUnit4, and it worked.  But, its a somewhat harder to read/debug.  If I come into a test that you wrote, I may not understand that you expected exception and thing its a bug in your test.  I would prefer that we were very explicit about when we expect an exception.  In JUnit4, I used code comments.  In JUnit5, there is no reason not to use the assertions.  It makes it very explicit as to what you intend/expect and others that follow you will have less chance of messing up your work.  One of the benefits of good tests is that YOU (the original author) don't have to be the one to go back and fix all the bugs.  You don't want to be the owner of something for all time.

Reply:
Not a problem.  Assertions added.